### PR TITLE
First pass fixing lint errors.

### PR DESCRIPTION
**Pull request by adamconnors**
Added pylint and cleaned up errors
Various other bits of tidy up, mostly driven by lint.

PTAL

### CODE REVIEW

File: server/agents/agents.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+"""Base class for all agents."""
```
Comment (alalamav) [inline server/agents/agents.py:1]: file-level comments should be below the copyright header

File: server/agents/default_agent_test.py [hunk 2]
```diff
@@ -12,14 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# pylint: disable=missing-module-docstring, missing-function-docstring, missing-class-docstring, line-too-long
```
Comment (alalamav) [inline server/agents/default_agent_test.py:15]: is this necessary?

**Review by alalamav — COMMENTED** [review]
are these errors coming from black? were the fixes automated or manual?

File: server/agents/agents.py (hunk 1 above)
Comment (adamconnors) [reply server/agents/agents.py:1]: will do -- Duckie led me astray

File: server/agents/default_agent_test.py (hunk 2 above)
Comment (adamconnors) [reply server/agents/default_agent_test.py:15]: My thinking is to have zero pylint errors ... If there's a bunch in tests that we just ignore then we'll never see when real ones crop up?

**Comment by adamconnors** [comment]
The were errors from pylint not black.
I fixed manually... but it syntax highlights so once done I imagine it'll stay fixed.

**Comment by adamconnors** [comment]
Finished first pass. PTAL.

(Will add return types separately)

File: README.md [hunk 3]
```diff
@@ -43,8 +43,35 @@ npm run start
...
 
-## Agent development
+## Run tests
+**WARNING**: Server scripts make calls to the language-model and will incur charges.
+
+### To run server tests from the **project root**:
+
+```sh
+# All tests
+python -m unittest discover -t . -s server -p "*_test.py"
```
Comment (alalamav) [inline README.md:53]: nit: should this be python3?

File: server/agents/history_tutor/history_tutor_test.py [hunk 4]
```diff
@@ -11,19 +11,20 @@
...
 import json
 import unittest
 
+from langchain_google_vertexai import VertexAI
+
 from ..agents import AgentState
 from .history_tutor import (
     HistoryTutor,
-    load_file,
+    _load_file,
```
Comment (alalamav) [inline server/agents/history_tutor/history_tutor_test.py:24]: `load_file` should not have an underscore if it's exported

**Review by alalamav — APPROVED** [review]
looks good! just a couple of questions

File: README.md (hunk 3 above)
Comment (adamconnors) [reply README.md:53]: fixed

File: server/agents/history_tutor/history_tutor_test.py (hunk 4 above)
Comment (adamconnors) [reply server/agents/history_tutor/history_tutor_test.py:24]: I wasn't sure because it's only called by the test?
I've removed _ for now anyway.

**Pull request merged**